### PR TITLE
Updates, and features

### PR DESCRIPTION
**Pull request by siteshackinc**
Updated Angular 1.4, ui-bootstrap v1, added a few extras like summernote, ui-datetime-picker, update all package.json & bower.json
Everything tested fine. Only issue I had was I had to npm cache clear before I npm install

### CODE REVIEW

**Comment by ilanbiala** [comment]
Thanks for the PR! A few things:

1. ui-bootstrap was already updated in a previous PR
2. We don't want to add UI features at this time, so we won't need the summernote and ui-datetime-picker components
3. What was there to update in the bower and package files?

File: modules/articles/client/views/form-article.client.view.html [hunk 1]
```diff
@@ -13,8 +13,8 @@
           </div>
         </div>
         <div class="form-group">
-          <label class="control-label" for="content">Content</label>
-          <textarea name="content" data-ng-model="vm.article.content" id="content" class="form-control" cols="30" rows="10" placeholder="Content"></textarea>
+          <label for="content">Content</label>
+          <summernote name="content" config="summeroptions" ng-model="vm.article.content" id="content"></summernote>
```
Comment (jloveland) [inline modules/articles/client/views/form-article.client.view.html]: I like the addition of summernote!

**Comment by siteshackinc** [comment]
The enhancements I did I know others like and wanted too. But authors don't want it period. Ill create a new public branch with all the changes I've done, if you want to use them that's great. If you have other things you would like, let me know, I'd be open vs being closed. I will also work on my material-meanjs branch for those that would like that too. Ill merge to it any updates or changes they do to the master to keep up with anything they do. enhanced-meanjs will be that repository.

File: modules/articles/client/views/form-article.client.view.html (hunk 1 above)
Comment (ilanbiala) [inline modules/articles/client/views/form-article.client.view.html]: We don't want to make our UI bloated and opinionated when most people aren't using MEAN for its UI. As I've said in other PRs, our focus at this time is not UI, so we don't want to put effort into that part of the framework right now.

**Comment by ilanbiala** [comment]
@siteshackinc can you rebase and fix up the PR so we can merge in some of this?

File: modules/articles/client/views/form-article.client.view.html (hunk 1 above)
Comment (lirantal) [inline modules/articles/client/views/form-article.client.view.html]: Agree with @ilanbiala on the UI effort argument. Although Ilan, if this is just minor addition and not really something that will introduce problems/maintenance overhead then I don't mind.

**Comment by lirantal** [comment]
@siteshackinc can you please:
1. provide a better description for this PR than 'updates and features'? I see that you've updated versions but there are also model updates like https://github.com/meanjs/mean/pull/1171/files#diff-240057bbb9109acf3b1745506df4b781 which includes new fields which I don't know if we care about (I personally don't mind too much but not sure it's worth to have them in meanjs framework as it will require more testing and maintenance)
2. Re-base and update this PR as it is currently with many commits and have merge conflicts.

File: package.json [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
-  "meanjs-version": "0.4.2",
+  "version": "0.4.3",
+  "meanjs-version": "0.4.3",
   "private": false,
   "author": "https://github.com/meanjs/mean/graphs/contributors",
   "license": "MIT",
   "repository": {
     "type": "git",
-    "url": "https://github.com/meanjs/mean.git"
+    "url": "https://github.com/siteshackinc/mean.git"
```
Comment (lirantal) [inline package.json]: This is wrong, can you apply the correct URL here again?

File: package.json [hunk 3]
```diff
@@ -1,14 +1,14 @@
 {
   "name": "meanjs",
   "description": "Full-Stack JavaScript with MongoDB, Express, AngularJS, and Node.js.",
-  "version": "0.4.2",
-  "meanjs-version": "0.4.2",
+  "version": "0.4.3",
+  "meanjs-version": "0.4.3",
```
Comment (lirantal) [inline package.json]: @ilanbiala is it required with this change that we increment the meanjs version? or can this version increment wait?

**Comment by lirantal** [comment]
@siteshackinc no interest in following this through I guess?
@ilanbiala would you like to continue where this PR left off and let's try to be specific only for some of the items in this PR?

**Comment by jloveland** [comment]
@siteshackinc I really like where you were going with these enhancements..any way you can come back with smaller PRs?

**Pull request closed without merging**